packages/@coorpacademy-components/src/atom/font-awsome-adaptive-icon/index.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {fas} from '@fortawesome/pro-solid-svg-icons';
+import {library} from '@fortawesome/fontawesome-svg-core';
+import toLower from 'lodash/fp/toLower';
+import merge from 'lodash/fp/merge';
+import getOr from 'lodash/fp/getOr';
+import get from 'lodash/fp/get';
+
+import style from './style.css';
+
+library.add(fas);
+
+const DEFAULT_SIZE = 'm';
+const DEFAULT_FA_SIZE = 'lg';
+const DEFAULT_WRAPPER_SIZE = 40;
+
+const COLOR_MAPPING = {
+  '#FFDCD1': '#A32700',
+  '#FFF9D1': '#A38300',
+  '#FAD6DE': '#93102F',
+  '#D9F4F7': '#1B7B88',
+  '#DDD1FF': '#2B00A3'
+};
+
+const SIZE_MAPPING = {
+  s: {
+    faSize: 'xs',
+    wrapperSize: 32
+  },
+  m: {
+    faSize: DEFAULT_FA_SIZE,
+    wrapperSize: DEFAULT_WRAPPER_SIZE
+  },
+  xl: {
+    faSize: 'xl',
+    wrapperSize: 48
+  }
+};
+
+const FontAwesomeAdaptiveIcon = ({
```

Comment:
Il faut que le nom soit générique. Pas besoin de mentionner dans le nom l'implementation

```suggestion
const Icon = ({
```